### PR TITLE
fix(sim-events-iracing): re-fire session-start callout on session change (#564)

### LINKED ISSUE #564: bug(sim-events-iracing): session-start callout doesn't re-fire after session change
### What happened?

The Race Engineer "session start" callout added in #542 / #551 plays the first time the driver goes on track, but stays silent for every subsequent session in the same iRacing instance. After a practice session ends and qualifying / race begins, the brief never plays again.

### Expected behavior

The callout fires once per session — practice, qualifying, and race each get their own session-start brief (greeting, session type, pit speed, track + air temps, wetness). This matches the wording in #542 ("fires exactly once per session") and in the #551 PR description ("Garage -> track -> garage -> track does not re-fire (per-session by design)").

### Steps to reproduce

1. Enter a multi-session iRacing event (practice + qualifying + race).
2. Go on track in practice — the session-start callout plays. ✅
3. Wait for practice to end and qualifying to start, go on track — **no callout**. ❌
4. Wait for race to start, go on track — **no callout**. ❌

### Root cause

`packages/sim-events-iracing/src/translator.ts:62-69` — `firstOnTrackSeeded` and `firstOnTrackFired` live on the `TranslatorInstance` (not on `TranslatorState`) and reset only in `handleDisconnect()` (lines 265-266). Once `firstOnTrackFired = true` in practice, it stays true for the rest of the iRacing connection, so `driver.firstOnTrack` never publishes again. The session-start scenario subscribes to that single event and is gated entirely by it.

The flags were placed on the instance to survive the replay-guard's per-tick state resets (correct for replay scrubbing), but the design accidentally also makes them survive session changes (wrong for the once-per-session contract).

### Fix outline

Add a session-change detector at the top of `handleTick` that runs before `diffFirstOnTrack`. On `telemetry.SessionNum` change:

- Re-create `state` from `createInitialState()`, preserving `lastSessionNum` so `diffLifecycle` still emits `session.changed` on the same tick.
- Reset `firstOnTrackFired = false` (keep `firstOnTrackSeeded = true` so the next tick doesn't re-seed silently and swallow the fire).

`packages/sim-events-iracing/src/diff/laps.ts:85-90` is the existing precedent — it already detects `sessionNum !== lastLapSessionNum` and wipes its own lap-completed tracking. The fix generalizes that pattern to the whole translator. `pitSpeedLimitMps` already keys by `${TrackID}|${SessionNum}` (translator.ts:465) so it auto-invalidates per session and needs no special handling. Every other diff uses the `*Initialized = false` seed-and-bail pattern and re-seeds cleanly from a fresh state.

### Affected packages

- `@iracedeck/sim-events-iracing` — the fix
- No other packages need changes; `audio-scenarios`, `event-bus`, `deck-core`, and both plugins all read live state and continue working unchanged.

### Documentation / website / skills

- None — this is a behavior fix, no documented contract changes.

### Verification

- New tests in `translator.test.ts` for: `driver.firstOnTrack` re-fires on session change (both mid-drive and garage-to-track), `session.changed` still emits, representative per-session state field resets.
- Existing test "fires driver.firstOnTrack only once across a car → exit → car cycle" (translator.test.ts:1173-1187) stays valid — that's within one session.
- Manual: practice → qualifying → race in a single iRacing event should produce three separate session-start callouts.

### iRacing version

Not version-specific — present since #551 was merged.

### Stream Deck model and software version

Not platform-specific — bug is in the shared sim-events translator and affects both Elgato and Mirabox builds.

### Additional context

Related: #542 (original feature request), #551 (implementing PR).

### PR DESCRIPTION
**Pull request by niklam**
## Related Issue

Fixes #564

## What changed?

`firstOnTrackFired` was placed on the translator instance with connection-lifetime semantics, so the Race Engineer session-start callout (added in #551 for #542) fired once per iRacing process — not once per session. Practice's fire silenced qualifying and the race.

Adds an instance-level `lastObservedSessionNum` tracker (outside the replay guard's state-wipe zone, so session transitions arriving on `IsReplayPlaying: true` ticks can't slip past) and a `resetPerSessionState` helper that runs at the top of `handleTick`. On a `SessionNum` delta the helper re-creates `state` from `createInitialState()` and clears `firstOnTrackFired`, while preserving the lifecycle diff's baselines (`lifecycleInitialized`, `lastSessionNum`, `lastEngineRunning`, `lastLap`) so `session.changed` still publishes for external subscribers and the reset doesn't synthesize spurious `engine.startup` / `lap.started`. `firstOnTrackSeeded` stays `true` — re-seeding would silently mark fired when the driver is already on track at the transition and swallow the very fire we want.

No other packages need changes. `audio-scenarios`, `event-bus`, `deck-core`, and both plugins continue to work unchanged because they subscribe to the bus event, not the instance state.

## How to test

1. `pnpm install && pnpm build && pnpm test && pnpm lint && pnpm format` — all green.
2. **iRacing:** join a practice + qualifying + race event. Go on track in each session. The Race Engineer session-start brief plays three separate times (greeting, session type, pit speed, temps, wetness) — once per session.
3. **Regression guard:** within a single session, garage → track → garage → track does NOT re-fire (still once-per-session).
4. **Tests cover:**
   - mid-drive `SessionNum` increment re-fires `driver.firstOnTrack`
   - `SessionNum` transition that arrives during an `IsReplayPlaying: true` tick still produces the fire on the next live-on-track tick
   - `session.changed` still publishes on the reset tick
   - representative per-session diff state (`carControl.p2pToggled` baseline) wipes so the next session's rising edge fires fresh

## Checklist

- [x] Linked to an approved issue
- [x] New code has unit tests
- [x] Changes registered in all applicable plugins (Stream Deck, Mirabox) — translator is shared by both; no per-plugin changes needed
- [x] No unrelated changes included

## Summary by CodeRabbit

* **Bug Fixes**
  * driver.firstOnTrack now reliably re-fires when SessionNum changes (practice→qualifying→race) and across replay→live transitions
  * session.changed is published on the same tick that triggers the session reset
  * radar.changed emits the proper teardown (left→clear) before per-session wipe
  * Per-session state is reset so events can re-trigger correctly in new sessions

* **Tests**
  * Added regression tests for session transitions, replay/live edge cases, radar teardown, and per-session state wiping (p2p toggle re-fire)

[Review Change Stack](https://app.coderabbit.ai/change-stack/niklam/iracedeck/pull/565?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: packages/sim-events-iracing/src/translator.ts [hunk 1]
```diff
@@ -259,13 +273,54 @@ function handleDisconnect(self: TranslatorInstance): void {
...
+    lifecycleInitialized: self.state.lifecycleInitialized,
+    lastSessionNum: self.state.lastSessionNum,
+    lastEngineRunning: self.state.lastEngineRunning,
+    lastLap: self.state.lastLap,
+  };
+
+  self.state = createInitialState();
+  Object.assign(self.state, preservedLifecycle);
+
+  self.firstOnTrackFired = false;
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/sim-events-iracing/src/translator.ts]: Fixed in `b475a988`. Went with the "emit the teardown before wiping" alternative (matches `handleDisconnect`'s existing pattern) rather than preserving `radarState`, so the new session genuinely starts with a clean radar slate instead of inheriting the prior session's beep state.

Also added a regression test that fires the radar on a practice tick and then transitions through a replay-mode session change, asserting the `radar.changed → clear` edge is observed before the wipe.
